src/swarm/neo/client/RetryTimer.d

```diff
@@ -98,8 +85,14 @@ body
         // Constructor; sets the timer to the next delay that is greater
         // than the duration of the first retry, `t_end - t_start`, and
         // registers this instance with epoll.
-        this ( )
+        this ( timespec t_start, timespec t_end )
         {
+            // This exception is caught inside the scope of this function.
+            this.e = new TimerFD.TimerException;
```

Comment:
These objects are now heap allocated. (They were previously stack allocated.)

Reply:
Oh, this is going to be interesting. Means only way to avoid closure is to use something like Phobos `emplace` (https://dlang.org/library/std/conv/emplace.html#2) :/

Reply:
We do have `emplace` inside swarm (we allocate request handlers on stack using it): https://github.com/sociomantic-tsunami/swarm/blob/6fa8a821aa6cb3bfbe00c61b9652aa0df43d2a3d/src/swarm/neo/node/ConnectionHandler.d#L691

Reply:
(it's a bit primitive atm, though, as it doesn't accept constructor arguments)

Reply:
I think we should add few utils like that to ocean considering how widespread this may end up being

Reply:
Without looking at the code: afair, there's no real reason for this code to be all nests within nests, like it's currently written. Could it be refactored to avoid this problem completely?

Reply:
Yeah I am going to do it tomorrow

Reply:
@gavin-norman-sociomantic I looked at it again and the whole stack allocation thing looks rather pointless to me here. For example, `TimerFD` constructor will heap allocate exception instance internally - it doesn't matter that you create it as `scope t = new TimerFD`, there will be GC activity anyway.

Same applies for few other fields - and considering usually GC is "all or nothing" for our code, it doesn't look like this is optimized as it claims to be.

Reply:
Cool. Ideally without using nasty tricks like `emplace` ;)

Reply:
See updated version - I have simply moved class to global scope and made it refer scope-allocated stuff as reference captured by constructor.